Makefile

```diff
@@ -60,3 +60,10 @@ install:
 .PHONY: develop
 develop:
 	$(PYTHON) setup.py develop
+
+
+.PHONY: tag-and-release
+tag: ## create a tag in git. to run, do a 'make VERSION="version string" tag-and-release
```

Comment:
"tag" vs "tag-and-release"?

Looks good otherwise. I'm assuming this is based on what's in Parsl.

Reply:
Named it that way, because it does both tag followed by release. Are you recommending that I break it up into two separate entry points?

Reply:
We might if we need to tag separately of releasing. I think it is ok for now unless you have different use cases in mind.

Reply:
The `PHONY` target is named `tag-and-release`, the make target is named `tag` - shouldn't that be the same?

Reply:
Yep, that's a bug. Let me fix that in a different PR since this one got merged.

Reply:
It may be useful to address comments before merging in the future :)